pkg/virtctl/create/vm/vm_test.go

```diff
@@ -939,6 +942,183 @@ chpasswd: { expire: False }`
 			Entry("with CloudInitConfigDrive (explicit)", configDriveUserData, configDriveNetworkDataB64, setFlag(CloudInitFlag, CloudInitConfigDrive)),
 		)
 
+		DescribeTable("VM with access credentials with type ssh and method ga and with", func(params string, verifyFn func(machine *v1.VirtualMachine), extraArgs ...string) {
+			args := append([]string{
+				setFlag(AccessCredFlag, params),
+			}, extraArgs...)
+			out, err := runCmd(args...)
+			Expect(err).ToNot(HaveOccurred())
+			vm, err := decodeVM(out)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(vm.Spec.Template.Spec.AccessCredentials).To(ConsistOf(
+				v1.AccessCredential{
+					SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
+						Source: v1.SSHPublicKeyAccessCredentialSource{
+							Secret: &v1.AccessCredentialSecretSource{
+								SecretName: "my-keys",
+							},
+						},
+						PropagationMethod: v1.SSHPublicKeyAccessCredentialPropagationMethod{
+							QemuGuestAgent: &v1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
+								Users: []string{"myuser"},
+							},
+						},
+					},
+				},
+			))
+
+			verifyFn(vm)
+
+			// No inference possible in this case
+			Expect(vm.Spec.Instancetype).To(BeNil())
+			Expect(vm.Spec.Preference).To(BeNil())
+		},
+			Entry("default type and method, user from param, nocloud default", "src:my-keys,user:myuser", verifyNoCloudGAManageSSH),
```

Comment:
It would be nice if ginkgo had some test matrix functions, so we didn't need to check 30 test cases manually for typos.

Reply:
Tried to do it with SubTree tables, it becomes marginally better.